packages/asset-uploader/src/patch.ts

```diff
@@ -40,4 +44,31 @@ export const patchAssets = async (
   if (deletedAssetIds.length !== 0) {
     deleteAssets({ projectId, ids: deletedAssetIds }, context, assetClient);
   }
+
+  // add new assets found in patched version
+  const addedAssets: Asset[] = [];
+  for (const [assetId, asset] of patchedAssets) {
+    // skip stubbed assets
+    if (asset === undefined) {
+      continue;
+    }
+    if (assets.has(assetId) === false) {
+      addedAssets.push(asset);
+    }
+  }
+  if (addedAssets.length !== 0) {
+    await prisma.asset.createMany({
+      data: addedAssets.map((asset) => ({
+        id: asset.id,
+        projectId: asset.projectId,
+        name: asset.name,
```

Comment:
This name can be arbitrary label now in theory? Doesn't link to file in store any more?

(Just curious)

Reply:
It's foreign key for File table and used to address actual file in storage.

Reply:
Ah, I thought tables are linked by id because I saw this line, and thought that file id becomes asset id eventually

https://github.com/webstudio-is/webstudio-builder/blob/ad7fafe9cf915ec63721c9c1d522d8f4b37b5d5c/apps/builder/app/builder/shared/assets/use-assets.tsx#L191

Which I now realise wouldn't work because many assets can use the same file. So maybe it worth double checking the line above and how id gets passed from there.

Reply:
Yeah, maybe better rename to assetId

Reply:
Got it 👍